Source/WebKit/WebProcess/WebPage/IPCTestingAPI.cpp

```diff
@@ -349,7 +350,7 @@ class JSMessageListener final : public IPC::MessageObserver {
 
     WeakPtr<JSIPC> m_jsIPC;
     Type m_type;
-    JSContextRef m_context;
+    JSC::Weak<WebCore::JSDOMGlobalObject> m_globalObject;
```

Comment:
This is way better than the status quo, so I'll say r+.

Did you consider storing a strong reference here? In the IPC testing infrastructure, is the JSMessageListener thingy reachable from the webpage? If so, strong would be a retain cycle. But if not, strong might be correct.

Is it a requirement that we use the same global object that the webpage uses? Did you consider just allocating our own global object, separate from the webpage? Or is there a specific need to use objects and types that come from the webpage when we do our IPC testing?

Reply:
The potential benefit of solving those follow-up issues is that you'd get testing coverage in these cases, instead of returning early.

Reply:
Yes, preventing the global object from being garbage collected was what I tried to do first, but I realized it was wrong. These message listeners are reachable from the webpage.

I'm not completely sure if it would be okay to create our own global object. It does seem like some IPC decoding related things might break if the global object we used was separate from the webpage (see `JSMessageListener::jsDescriptionFromDecoder`), but I can't really explain what/why.